xbmc/addons/gui/skin/SkinTimerManager.cpp

```diff
@@ -166,14 +166,35 @@ void CSkinTimerManager::TimerStart(const std::string& timer) const
 
 void CSkinTimerManager::TimerStop(const std::string& timer) const
 {
-  if (m_timers.count(timer) == 0)
+  if (!TimerExists(timer))
   {
     CLog::LogF(LOGERROR, "Couldn't find Skin Timer with name: {}", timer);
     return;
   }
   m_timers.at(timer)->Stop();
 }
 
+size_t CSkinTimerManager::GetTimerCount() const
+{
+  return m_timers.size();
+}
+
+bool CSkinTimerManager::TimerExists(const std::string& timer) const
+{
+  return m_timers.count(timer) != 0;
+}
+
+std::unique_ptr<CSkinTimer> CSkinTimerManager::GrabTimer(const std::string& timer)
+{
+  if (!TimerExists(timer))
+  {
+    return nullptr;
+  }
+  auto timerInstance = std::move(m_timers.at(timer));
+  m_timers.extract(timer);
+  return timerInstance;
```

Comment:
This saves looking up the value three times:
```suggestion
  if (auto iter = m_timers.find(timer); iter != m_timers.end())
  {
    auto timerInstance = std::move(iter->second);
    m_timers.erase(iter);
    return timerInstance;
  }
  else
  {
    return {};
  }
```
Also using extract here is a bit unusual.

Reply:
Thanks, I applied your change (removing the else)